src/tool_operate.c

```diff
@@ -404,14 +404,8 @@ static CURLcode pre_transfer(struct GlobalConfig *global,
     }
 #endif
 
-    if(uploadfilesize != -1) {
-      struct OperationConfig *config = per->config; /* for the macro below */
-#ifdef CURL_DISABLE_LIBCURL_OPTION
-      (void)config;
-      (void)global;
```

Comment:
This leads to an unused parameter warning with `CURL_DISABLE_LIBCURL_OPTION` because the `my_setopt` macros don't use `global` in this case:
https://curl.se/dev/log.cgi?id=20250312061525-3878059#prob1

Reply:
I'm on it.

That would also indicate we could use a new CI job...

Reply:
#16684